### DISCUSSION TITLE
Vercel - NodeJS : Cron job configuration issue

### DISCUSSION THREAD

**Discussion by ShinanTc** [vercel/community#3749] [Help]
### Summary

I am using node js and below is my vercel.json file:

vercel.json:

```
{
    "version": 2,
    "builds": [
      {
        "src": "./index.js",
        "use": "@vercel/node"
      }
    ],
    "routes": [
      {
        "src": "/(.*)",
        "dest": "/"
      }
    ],
    "crons": [
      {
        "path": "/api/cron",
        "schedule": "0 10 * * *"
      }
    ]
  }
```

Below is my `/api/cron` route:

```
router.get("/api/cron", async (req, res) => {
  console.log("Cron job is working (FROM ROUTE)");
  res.send("Cron job is working");
});

```
The file path of this route is `/src/routes/stock/stock.js`. Is there any configuration missing or is there something wrong with what I am doing? Stuck here for almost a week. The logs are not getting printed out even when we trigger by clicking the "Run" button manually from settings -> Cron Jobs and the cron job isn't working as well.

FOR MORE REFERENCE - [vercel documentation/cron job](https://vercel.com/docs/cron-jobs)

### Example

[https://github.com/ShinanTc/stockRecommender](https://github.com/ShinanTc/stockRecommender)

### Steps to Reproduce

* Create a sample route and give the route as a path on vercel.json. 
* Provide the time you want to run the program on the schedule field.  
* Then run `npx vercel deploy --prod`
* Go to you vercel `project -> settings -> cron jobs`
* Try triggering it and it wont work (No logs appearing or not doing anything as expected

**Comment by amyegan** [comment]
Hey @ShinanTc. I have a couple of debugging ideas to try. 
- Have you verified that the `/api/cron` path exists and works on your deployment? You can visit it in the browser to check that it returns the expected value on a `GET` request.
- Did you check for the logs in the production deployment's [runtime logs](https://vercel.com/docs/observability/runtime-logs), or did you look in the browser console? If the route being called is `/api/cron`, then you will see `[GET] /api/cron` in the "Request" column of the logs table. You may need to refresh the results to get the latest logs after manually triggering the cron job.

**Reply by ShinanTc** [reply]
Hi @amyegan,

Thanks for your response.

* I've looked at the `/api/cron` route in my web browser, and it's there and working correctly.
* Also, I tried running it manually by triggering the "RUN" option inside vercel settings -> cron jobs. It is working fine from there as well (The logs were there). The cron job is not working automatically at the scheduled time.

**Reply by amyegan** [reply]
Did you check the logs at the scheduled time for your local time zone or for UTC time? Remember that the cron schedule uses UTC time and the [runtime log storage limit](https://vercel.com/docs/observability/runtime-logs#limits) is 1 hour for Hobby and Pro accounts.

**Reply by kilinkis** [reply]
I have the same issue. In the log I see an error 500 that happens only when the cron runs the function. 
The error ourput is not very clear. I'm investigating it now.

**Reply by Jasminder-Singh** [reply]
Hi , I was facing the same issue. The cron job did not run automatically untill you deploy your website or application in production mode. I fix this via go to my vercel dashboard and click on the project which I deploy recently and set to in production. After this all works fine. You can use this command in your terminal of your project -> vercel deploy --prod. This will deploy your web app to production mode.
This will definitely work

**Reply by Deamoner** [reply]
I am experiencing this issue. I hit the endpoint and get the response, so I know it's there and works. I know it's in my vercel.json and that is correct as it shows in my project settings. I even hit run on there. But none of these actions that show as working show in the logs. What is the issue?

**Reply by djshubs** [reply]
We have noticed an issue as well where our cron job is working in production but not on preview/staging branch.

**Reply by rsimonton** [reply]
https://vercel.com/guides/how-to-setup-cron-jobs-on-vercel#deploy-your-project

> Once your project is deployed on Vercel, our build process will create the cron job. **Cron jobs are invoked only for production deployments. Preview deployments are ignored.**
